src/modules/atommodel/primitives/RadixResourceIdentifier.ts

```diff
@@ -49,4 +53,8 @@ export class RadixResourceIdentifier implements RadixPrimitive {
 
         return encoder.pushAny(output)
     }
+
+    public equals(rri: RadixResourceIdentifier) {
+        return this.address === rri.address && this.unique === rri.unique && this.type === rri.type
```

Comment:
Since address is an object, `this.address === rri.address` will fail for 2 different instances of the same address. Use `address.equals()`